refactor(semantic): alter syntax of `control_flow!` macro

overlookmotel opened the pull request:
Previously:

```rs
let ix = control_flow!(|self, cfg| cfg.current_node_ix);
```

after this PR:

```rs
let ix = control_flow!(self, |cfg| cfg.current_node_ix);
```

It expands to:

```rs
let ix = if let Some(ref mut cfg) = self.cfg {
    cfg.current_node_ix
} else {
    Default::default()
};
```

So rationale for this change is that it makes it clearer that `self` is passed *in* and `cfg` comes *out* into the "closure".

overlookmotel commented:
@rzvxa Out of interest, why `if let Some(ref mut cfg) = self.cfg` rather than `if let Some(cfg) = &mut self.cfg`? Is there a subtle difference between the two?

rzvxa commented:
> @rzvxa Out of interest, why `if let Some(ref mut cfg) = self.cfg` rather than `if let Some(cfg) = &mut self.cfg`? Is there a subtle difference between the two?

Not a functional one, I find this too noisy(lots of symbols).

```rust
if let Some($cfg) = &mut $self.cfg {
    $body
} else {
}
```

overlookmotel commented:
Thanks for satisfying my curiosity!

overlookmotel committed:
refactor(semantic): alter syntax of `control_flow!` macro (#4275)

Previously:

```rs
let ix = control_flow!(|self, cfg| cfg.current_node_ix);
```

after this PR:

```rs
let ix = control_flow!(self, |cfg| cfg.current_node_ix);
```

It expands to:

```rs
let ix = if let Some(ref mut cfg) = self.cfg {
    cfg.current_node_ix
} else {
    Default::default()
};
```

So rationale for this change is that it makes it clearer that `self` is passed *in* and `cfg` comes *out* into the "closure".
